packages/alfa-rules/src/sia-r11/rule.ts

```diff
@@ -48,7 +48,11 @@ export default Rule.Atomic.of<Page, Element>({
 });
 
 export namespace Outcomes {
-  export const HasName = Ok.of("The link has an accessible name");
+  export const HasName = Ok.of(
+    Diagnostic.of(`The link has an accessible name`)
+  );
 
-  export const HasNoName = Err.of("The link does not have an accessible name");
+  export const HasNoName = Err.of(
+    Diagnostic.of(`The link does not have an accessible name`)
+  );
```

Comment:
The need to pile up `Ok.of(Diagnostic.of(…))` (or `Err.of(Diagnostic.of(…))`) is a bit boilerplating and annoying. But it's not clear we can do much better…

maybe a `foo(outcome: "passed" | "failed", msg, ...args): Result<Diagnostic>` could help, but it's not clear (and it won't really be useful for subclasses of `Diagnostic` anyway)…

Reply:
> (and it won't really be useful for subclasses of Diagnostic anyway)…

That's precisely why we really can't make this any shorter than it already is. I'm not too worried though to be honest, as I do like the clarity it provides. As mentioned previously, abstraction is a worthy pursuit up to a point, but we shouldn't forget the value of composability.